app/jobs/enforce_timeouts_job.rb

```diff
@@ -1,20 +1,30 @@
+# The EnforceTimeoutsJob searches for BuildAttempts that were picked up by a
+# kochiku worker but never heard back from again. It compares (Time.now -
+# started_at) against the timeout value of the repository. If the maximum time
+# has elapsed, it will mark the BuildAttempt as errored and kick off a rebuild.
 class EnforceTimeoutsJob
   def self.perform
-    BuildAttempt.where(:state => 'running').each do |attempt|
-      lenient_timeout = attempt.build_instance.repository.timeout + 15
+    # The EnforceTimeoutsJob runs frequently so we do not check BuildAttempts greater than 1 day old
```

Comment:
I agree that 1 day is overkill but I wanted to have a healthy buffer. With the new index on created_at the performance should be good.